core/src/proxy/fetch_stage_manager.rs

```diff
@@ -61,97 +73,104 @@ impl FetchStageManager {
     ///      Sets fetch_connected to true, pending_disconnect to false
     ///      Advertises saved contact info
     fn start(
-        cluster_info: Arc<ClusterInfo>,
-        heartbeat_rx: Receiver<HeartbeatEvent>,
-        packet_intercept_rx: Receiver<PacketBatch>,
-        packet_tx: Sender<PacketBatch>,
-        exit: Arc<AtomicBool>,
-    ) -> JoinHandle<()> {
-        Builder::new().name("fetch-stage-manager".into()).spawn(move || {
-            let my_fallback_contact_info = cluster_info.my_contact_info();
+        global_relayer_config: &Arc<Mutex<RelayerConfig>>,
+        cluster_info: &Arc<ClusterInfo>,
+        heartbeat_rx: &Receiver<HeartbeatEvent>,
+        packet_intercept_rx: &Receiver<PacketBatch>,
+        packet_tx: &Sender<PacketBatch>,
+        exit: &Arc<AtomicBool>,
+    ) -> ProxyResult<()> {
+        let my_fallback_contact_info = cluster_info.my_contact_info();
```

Comment:
can you add a comment on what this is and why we don't care if `cluster_info` gets changed via the admin rpc?

Reply:
added a comment on the contact info. not sure what you would like to include wrt not caring if cluster_info gets changed

Reply:
So let's say validator starts up advertising relayer as it's TPU. Now let's say the following happens:
1. Relayer crashes and misses heartbeat
2. Validator falls back to original tpu
3. Validator changes tpu socket via admin rpc
4. Relayer reconnects
5. Fails heartbeat
6. Validator falls back to the original socket even though they had changed it in step 3